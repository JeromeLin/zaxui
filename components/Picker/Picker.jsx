import React, { Component } from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';
import { arrayTreeFilter, formatToInit, formatBackToObject } from './utils';
import ColumnGroup from './ColumnGroup';
import Cascader from './Cascader';
import Popup from '../Popup';


// 阻止选择器区域的默认事件
function onContainerClick(e) {
  e.stopPropagation();
}

class Picker extends Component {
  constructor(props) {
    super(props);

    let _data = null;
    let _value = null;

    if (Object.prototype.toString.call(props.dataSource[0]) !== '[object Array]' && !Object.prototype.hasOwnProperty.call(props.dataSource[0], 'children')) {
      _data = [props.dataSource];
      _value = props.value ? [props.value] : [];
    } else {
      _data = props.dataSource;
      _value = props.value;
    }

    this.state = {
      visible: props.visible || false,
      value: _value,
      data: _data,
      cascade: Object.prototype.toString.call(props.dataSource[0]) !== '[object Array]' && Object.prototype.hasOwnProperty.call(props.dataSource[0], 'children'),
    };

    this.tempValue = _value;
  }

  componentWillReceiveProps(nextProps) {
    if ('value' in nextProps) {
      let _value = null;
      let _data = null;


      if (Object.prototype.toString.call(nextProps.dataSource[0]) !== '[object Array]' && !Object.prototype.hasOwnProperty.call(nextProps.dataSource[0], 'children')) {
        _value = nextProps.value ? [nextProps.value] : [];
        _data = [nextProps.dataSource];
      } else {
        _value = nextProps.value;
        _data = nextProps.dataSource;
      }

      this.setState({
        value: _value,
        data: _data,
        cascade: Object.prototype.toString.call(nextProps.dataSource[0]) !== '[object Array]' && Object.prototype.hasOwnProperty.call(nextProps.dataSource[0], 'children'),
      });
    }
  }

  onValueChange(value) {
    this.setState({
      value,
    });
    this.props.onChange(value);
  }

  onCancel() {
    const { onCancel } = this.props;
    this.toggle();
    this.setState({
      value: this.tempValue,
    });
    onCancel && onCancel();
  }

  onOk() {
    const { onOk, valueMember, cols } = this.props;
    const { data, cascade } = this.state;
    const value = this.getInitValue();
    this.tempValue = value;
    this.toggle();
    let _value = null;
    // _value = value.length === 1 ? value.toString() : value;
    _value = formatBackToObject(data, value, cascade, valueMember, cols);
    onOk && onOk(_value);
  }

  onMaskClick() {
    const { onMaskClick } = this.props;
    this.onCancel();
    onMaskClick && onMaskClick();
  }

  getInitValue() {
    const data = this.state.data;
    const { valueMember } = this.props;

    const { value } = this.state;

    if (!value || !value.length) {
      if (this.state.cascade) {
        return formatToInit(data[0], valueMember, this.props.cols);
      }
      return data.map(d => (d[0][valueMember]));
    }

    return value;
  }

  // 切换显示状态
  toggle() {
    if (this.props.disabled) {
      return;
    }
    this.setState({
      visible: !this.state.visible,
    });
  }

  handleClick() {
    this.props.onClick();
    !this.props.disabled && this.toggle();
  }

  close(key) {
    this.setState({
      [`${key}`]: false,
    });
  }

  render() {
    const { prefixCls, format, disabled, className, cancelText, okText, title, placeholder, displayMember, valueMember } = this.props;
    const { data, value } = this.state;

    let PickerCol = null;

    const classes = classnames({
      [`${prefixCls}-container`]: true,
      [className]: !!className,
    });

    const inputCls = classnames({
      [`${prefixCls}-input`]: true,
      [`${prefixCls}-placeholder`]: !value.join(format),
      [`${prefixCls}-disabled`]: !!disabled,
    });

    const cols = data.map((d) => {
      return { props: { children: d } };
    });

    if (this.state.cascade) {
      PickerCol = (
        <Cascader
          prefixCls={prefixCls}
          data={data}
          value={this.state.value}
          cols={this.props.cols}
          displayMember={displayMember}
          valueMember={valueMember}
          onChange={v => this.onValueChange(v)}
          />
      );
    } else {
      PickerCol = (
        <ColumnGroup
          className={className}
          prefixCls={prefixCls}
          displayMember={displayMember}
          valueMember={valueMember}
          selectedValue={value}
          onValueChange={v => this.onValueChange(v)} >
          {cols}
        </ColumnGroup>
      );
    }

    const display = () => {
      if (this.state.cascade) {
        if (value.length) {
          const treeChildren = arrayTreeFilter(this.props.dataSource, (item, level) => {
            return item[valueMember] === value[level];
          });

          return treeChildren.map((v) => {
            return v[displayMember];
          }).join(format);
        }

        return value.join(format) || placeholder;
      }

      let treeChildren2 = this.props.dataSource.reduce((a, b) => {
        return a.concat(b);
      }, []);

      treeChildren2 = treeChildren2.filter((item) => { return ~value.indexOf(item[valueMember]); });

      return treeChildren2.map((v) => {
        return v[displayMember];
      }).join(format) || placeholder;

      // return value.join(format) || placeholder;
    };

    return (
      <div className={`${prefixCls}`} onClick={() => this.handleClick()}>
        <div className={inputCls}>
          <input type="hidden" value={this.state.value} />
          {display()}
        </div>
        <div className={classes} onClick={e => onContainerClick(e)}>
          <Popup
            className="za-popup-inner"
            visible={this.state.visible}
            onMaskClick={() => this.onMaskClick()}>
            <div className={`${prefixCls}-wrapper`}>
              <div className={`${prefixCls}-header`}>
                <div className={`${prefixCls}-cancel`} onClick={() => this.onCancel()}>{cancelText}</div>
                <div className={`${prefixCls}-title`}>{title}</div>
                <div className={`${prefixCls}-submit`} onClick={() => this.onOk()}>{okText}</div>
              </div>
              <div className={`${prefixCls}-mask-top`}>
                <div className={`${prefixCls}-mask-bottom`}>
                  {PickerCol}
                </div>
              </div>
            </div>
          </Popup>
        </div>
      </div>
    );
  }
}

Picker.propTypes = {
  visible: PropTypes.bool,
  placeholder: PropTypes.string,
  title: PropTypes.string,
  cancelText: PropTypes.string,
  okText: PropTypes.string,
  format: PropTypes.string,
  disabled: PropTypes.bool,
  dataSource: PropTypes.arrayOf(PropTypes.oneOfType([PropTypes.object, PropTypes.array])).isRequired,
  cols: PropTypes.number,
  onOk: PropTypes.func.isRequired,
  onCancel: PropTypes.func,
  onMaskClick: PropTypes.func,
  prefixCls: PropTypes.string,
  displayMember: PropTypes.string,
  valueMember: PropTypes.string,
};

Picker.defaultProps = {
  visible: false,
  placeholder: '请选择',
  title: '请选择',
  cancelText: '取消',
  okText: '确定',
  format: '',
  disabled: false,
  dataSource: [],
  value: [],
  onClick: () => {},
  onChange: () => {},
  onOk: () => {},
  onCancel: () => {},
  onMaskClick: () => {},
  prefixCls: 'za-picker',
  displayMember: 'label',
  valueMember: 'value',
};

export default Picker;
