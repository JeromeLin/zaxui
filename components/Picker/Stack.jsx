import React, { Component, PropTypes } from 'react'
import classnames from 'classnames'

class PickerStack extends Component {
  constructor(props) {
    super(props);

    this.state = {
      visible: false,
      value: props.value,
      errorMsg: ''
    };
  }

  componentDidMount() {
    this.reposition();
  }

  componentWillReceiveProps({ value, validate }) {
    this.setState({ value, errorMsg: validate(value) });
  }

  componentDidUpdate() {
    this.reposition();
  }

  change(index, cVal) {
    let { cols, validate, onOk } = this.props,
      value = this.state.value.slice(0, index),
      errorMsg;

    cVal && (value[index] = cVal);
    errorMsg = validate(value);

    if (index === cols - 1 && !errorMsg) {
      this.hide();
      onOk(value);
    } else {
      this.setState({ value, errorMsg });
    }
  }

  show() {
    let { value, validate } = this.props;

    this.setState({ visible: true, value, errorMsg: validate(value) });
  }

  hide() {
    this.setState({ visible: false });
  }

  stopPropagation(e) {
    e.stopPropagation();
    e.nativeEvent.stopImmediatePropagation();
  }

  labelCompile(data) {
    let { labelCompile } = this.props;

    if (data && typeof labelCompile === 'function') {
      return labelCompile(data);
    } else {
      return '';
    }
  }

  reposition() {
    let { dataSource, valueMember, disabled, displayItem, itemHeight } = this.props;

    if (disabled) return;

    this.state.value.reduce((data, item, index) => {

      let value = item[valueMember],
        valIndex = data.map(dataItem => dataItem[valueMember]).indexOf(value);

      if (~valIndex) {
        let target = this.refs[`column${ index }`],
          position = target.scrollTop;

        if (position < ((valIndex - displayItem + 1) * itemHeight) || position > (valIndex * itemHeight)) {
          target.scrollTop = valIndex * itemHeight;
        }

        return data[valIndex].children || [];
      }

      return [];

    }, dataSource)
  }

  render() {
    const { className, value: bVal, title, dataSource, placeholder, disabled, crumbAddon } = this.props,
      { visible, errorMsg, value } = this.state,
      labelItem = this.labelCompile(bVal[bVal.length - 1]),
      labelCls = classnames({
        'ui-picker-placeholder': !labelItem,
        'ui-picker-disabled': disabled
      }),
      wrapperCls = classnames({
        'ui-picker-container': true,
        'ui-picker-hidden': !visible,
        [className]: !!className
      });

    return (
      <div className="ui-picker">
        <div className={ labelCls } onClick={ () => !disabled && this.show() }>{ labelItem || placeholder }</div>
        {
          disabled ?
          null :
          <div className={ wrapperCls }>
            <div className="ui-picker-mask" onClick={ () => this.hide() }></div>
            <div className="ui-picker-inner">
              <div className="ui-picker-header">
                <div className="ui-picker-cancel" onClick={ () => this.hide() }>取消</div>
                <div className="ui-picker-title">{ title }</div>
                <div className="ui-picker-submit"></div>
              </div>
              <div className="ui-picker-crumbs">
                <p>选择：{ value.map((item, index) => <span key={ index } className="ui-picker-crumb" data-crumb-addon={ crumbAddon }>{ this.labelCompile(item) }</span>) }</p>
                { errorMsg ? <p className="ui-picker-error">{ errorMsg }</p> : null }
              </div>
              <div className="ui-picker-stack-group">{ this.renderGroup(dataSource, value) }</div>
            </div>
          </div>
        }
      </div>
    )
  }

  renderGroup(dataSource, value) {
    let { valueMember, cols } = this.props,
      group = [],
      i = 0;

    while(dataSource && i < cols) {
      let colVal = value[i];

      group.push(this.renderColumn(dataSource, i, colVal || { [valueMember]: '' }));

      dataSource = ((colVal ? dataSource.filter(item => item[valueMember] === colVal[valueMember])[0] : dataSource[0]) || {}).children;

      i++;
    }

    return group;
  }

  renderColumn(list, colIndex, data) {
    let { valueMember } = this.props,
      sVal = data[valueMember],
      cls = classnames({
        'ui-picker-stack-column': true,
        'lower-hidden': !sVal
      });

    return (
      <div
        key={ colIndex }
        className={ cls }
        onClick={ () => this.change(colIndex - 1) }
      >
        <div
          className="ui-picker-stack-column-wrapper"
          ref={ `column${ colIndex }` }
          onClick={ this.stopPropagation }
        >
          {
            list.map((item, index) => (
              <div
                key={ index }
                className={
                  classnames({
                    'ui-picker-stack-item': true,
                    'active': item[valueMember] === sVal
                  })
                }
                onClick={ () => this.change(colIndex, item) }
              >
                { this.labelCompile(item) }
              </div>
            ))
          }
        </div>
      </div>
    )
  }
}

PickerStack.propTypes = {
  value: PropTypes.array,
  valueMember: PropTypes.string,
  title: PropTypes.string,
  placeholder: PropTypes.string,
  disabled: PropTypes.bool,
  dataSource: PropTypes.array,
  cols: PropTypes.number,
  crumbAddon: PropTypes.string,
  displayItem: PropTypes.number,
  itemHeight: PropTypes.number,
  onOk: PropTypes.func,
  labelCompile: PropTypes.func,
  validate: PropTypes.func
}

PickerStack.defaultProps = {
  value: [],
  valueMember: 'value',
  title: '',
  placeholder: '',
  disabled: false,
  dataSource: [],
  cols: 3,
  crumbAddon: ' > ',
  displayItem: 8,
  itemHeight: 30,
  onOk: () => {},
  labelCompile: data => data.label,
  validate: data => ''
}

export default PickerStack
