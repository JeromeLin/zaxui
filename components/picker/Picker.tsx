import React, { PureComponent } from 'react';
import classnames from 'classnames';
import Popup from '../popup';
import PickerView from '../picker-view';
import BasePickerProps from './PropsType';
import parseProps from '../picker-view/utils/parseProps';

export interface PickerProps extends BasePickerProps {
  prefixCls?: string;
  className?: any;
}

export type DataSource = Array<{ [key: string]: any; children?: DataSource }>;

export interface PickerState {
  value: string[] | number[];
  objValue?: string[] | number[];
  dataSource: DataSource;
  visible: boolean;
  tempObjValue?: string[] | number[];
  tempValue?: string[] | number[];
}

export default class Picker extends PureComponent<PickerProps, PickerState> {
  static defaultProps = {
    dataSource: [],
    prefixCls: 'za-picker',
    valueMember: 'value',
    cols: Infinity,
    itemRender: data => data.label,
    visible: false,
  };

  private isScrolling = false;

  state: PickerState = parseProps.getSource(this.props);

  componentWillReceiveProps(props) {
    const state: PickerState = parseProps.getSource(props);
    state.tempValue = state.value;
    state.tempObjValue = state.tempObjValue;
    this.setState(state);
  }

  onChange = (selected) => {
    const { valueMember, onChange } = this.props;
    const value = selected.map(item => item[valueMember!]);
    this.setState({
      value,
      objValue: selected,
    });
    if (typeof onChange === 'function') {
      onChange(selected);
    }
  };

  onCancel = () => {
    const { onCancel } = this.props;
    const { tempValue = [], tempObjValue = [] } = this.state;
    this.setState({
      value: tempValue,
      objValue: tempObjValue,
      visible: false,
    });
    if (typeof onCancel === 'function') {
      onCancel();
    }
  };

  onOk = () => {
    if (this.isScrolling) {
      return false;
    }
    const { value, objValue = [] } = this.state;
    this.setState({
      value,
      objValue,
      visible: false,
    });

    const { onOk } = this.props;
    if (typeof onOk === 'function') {
      onOk(objValue);
    }
  };

  onMaskClick = () => {
    const { onMaskClick } = this.props;
    if (typeof onMaskClick === 'function') {
      onMaskClick();
    }
  };

  onTransition = (isScrolling) => {
    const { onTransition } = this.props;
    this.isScrolling = isScrolling;
    if (typeof onTransition === 'function') {
      onTransition(isScrolling);
    }
  };

  render() {
    const { prefixCls, className, cancelText, okText, title, children, locale, ...others } = this.props;
    const { visible, value } = this.state;
    const cls = classnames(prefixCls, className);
    return (
      <Popup
        visible={visible}
        onMaskClick={this.onMaskClick}
      >
        <div className={cls} onClick={(e) => { e.stopPropagation(); }}>
          <div className={`${prefixCls}__header`}>
            <div className={`${prefixCls}__cancel`} onClick={this.onCancel}>{cancelText || locale!.cancelText}</div>
            <div className={`${prefixCls}__title`}>{title || locale!.title}</div>
            <div className={`${prefixCls}__submit`} onClick={this.onOk}>{okText || locale!.okText}</div>
          </div>
          <PickerView
            {...others}
            visible={visible}
            value={value}
            onChange={this.onChange}
            onTransition={(isScrolling) => { this.onTransition(isScrolling); }}
          />
        </div>
      </Popup>
    );
  }
}
