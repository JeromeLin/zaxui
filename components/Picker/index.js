import Picker from './Picker';
import Date from './DatePicker';
import Stack from './Stack';

Picker.Date = Date;
Picker.Stack = Stack;

export default Picker;
