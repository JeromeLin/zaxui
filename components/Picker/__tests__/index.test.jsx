import React from 'react';
import { render, shallow, mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import Picker from '../index';

const District = [{
  value: '340000',
  label: '安徽省',
  children: [{
    value: '340800',
    label: '安庆市',
    children: [{
      value: '340803',
      label: '大观区',
      children: [],
    }],
  }],
}, {
  value: '820000',
  label: '澳门特别行政区',
  children: [{
    value: '820100',
    label: '澳门半岛',
    children: [],
  }, {
    value: '820200',
    label: '离岛',
    children: [],
  }],
}, {
  value: '110000',
  label: '北京',
  children: [{
    value: '110100',
    label: '北京市',
    children: [{
      value: '110114',
      label: '昌平区',
      children: [],
    }, {
      value: '110105',
      label: '朝阳区',
      children: [],
    }, {
      value: '110103',
      label: '崇文区',
      children: [],
    }, {
      value: '110115',
      label: '大兴区',
      children: [],
    }, {
      value: '110101',
      label: '东城区',
      children: [],
    }, {
      value: '110111',
      label: '房山区',
      children: [],
    }, {
      value: '110106',
      label: '丰台区',
      children: [],
    }, {
      value: '110108',
      label: '海淀区',
      children: [],
    }, {
      value: '110116',
      label: '怀柔区',
      children: [],
    }, {
      value: '110109',
      label: '门头沟区',
      children: [],
    }, {
      value: '110228',
      label: '密云县',
      children: [],
    }, {
      value: '110117',
      label: '平谷区',
      children: [],
    }, {
      value: '110230',
      label: '其它区',
      children: [],
    }, {
      value: '110107',
      label: '石景山区',
      children: [],
    }, {
      value: '110113',
      label: '顺义区',
      children: [],
    }, {
      value: '110112',
      label: '通州区',
      children: [],
    }, {
      value: '110102',
      label: '西城区',
      children: [],
    }, {
      value: '110104',
      label: '宣武区',
      children: [],
    }, {
      value: '110229',
      label: '延庆县',
      children: [],
    }],
  }],
},
];


describe('Picker', () => {
  it('Picker render visible', () => {
    const wrapper = render(
      <Picker
        dataSource={[
          { value: '1', label: '选项一' },
          { value: '2', label: '选项二' },
        ]}
        visible
        />
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should trigger onOk when press ok button', () => {
    const onOkFn = jest.fn();
    const onCancelFn = jest.fn();

    const wrapper = mount(
      <Picker
        dataSource={[
          {
            value: '1',
            label: '选项一',
            children: [
              { value: '11', label: '选项一' },
              { value: '12', label: '选项二' },
            ],
          },
          {
            value: '2',
            label: '选项一',
            children: [
              { value: '21', label: '选项一' },
              { value: '22', label: '选项二' },
            ],
          },
        ]}
        value={['1', '12']}
        onOk={onOkFn}
        onCancel={onCancelFn}
        />
    );

    wrapper.find('.za-picker-submit').simulate('click');
    expect(onOkFn).toBeCalled();
    expect(onCancelFn).not.toBeCalled();
  });

  it('should trigger onCancel when press cancel button', () => {
    const onOkFn = jest.fn();
    const onCancelFn = jest.fn();

    const wrapper = mount(
      <Picker
        dataSource={[
          {
            value: '1',
            label: '选项一',
            children: [
              { value: '11', label: '选项一' },
              { value: '12', label: '选项二' },
            ],
          },
          {
            value: '2',
            label: '选项一',
            children: [
              { value: '21', label: '选项一' },
              { value: '22', label: '选项二' },
            ],
          },
        ]}
        value={['1', '12']}
        onOk={onOkFn}
        onCancel={onCancelFn}
        />
    );

    wrapper.find('.za-picker-cancel').simulate('click');
    expect(onCancelFn).toBeCalled();
    expect(onOkFn).not.toBeCalled();
  });

  it('receive new dataSource', () => {
    const wrapper = shallow(
      <Picker
        dataSource={[
          { value: '1', label: '选项一' },
          { value: '2', label: '选项二' },
        ]}
        />
    );
    wrapper.setProps({
      dataSource: [
        { value: 'a', label: '选项一' },
        { value: 'b', label: '选项二' },
        { value: 'c', label: '选项三' },
      ],
    });
  });

  it('receive new value', () => {
    const wrapper = shallow(
      <Picker
        dataSource={[
          { value: '1', label: '选项一' },
          { value: '2', label: '选项二' },
        ]}
        />
    );
    wrapper.setProps({ value: '1' });
  });

  it('receive new cascader dataSource', () => {
    const wrapper = shallow(
      <Picker
        dataSource={[
          {
            value: '1',
            label: '选项一',
            children: [
              { value: '11', label: '选项一' },
              { value: '12', label: '选项二' },
            ],
          },
          {
            value: '2',
            label: '选项一',
            children: [
              { value: '21', label: '选项一' },
              { value: '22', label: '选项二' },
            ],
          },
        ]}
        />
    );

    wrapper.setProps({
      dataSource: [
        {
          value: '3',
          label: '选项一',
          children: [
            { value: '31', label: '选项一' },
            { value: '32', label: '选项二' },
          ],
        },
        {
          value: '4',
          label: '选项一',
          children: [
            { value: '41', label: '选项一' },
            { value: '42', label: '选项二' },
          ],
        },
      ],
    });
  });

  it('StackPicker', () => {
    jest.useFakeTimers();
    const wrapper = mount(
      <Picker.Stack
        dataSource={District}
        />
    );
    expect(toJson(wrapper)).toMatchSnapshot();
    wrapper.setProps({ value: ['安徽省', '安庆市', '大观区'] });
    jest.runAllTimers();
    wrapper.unmount();
  });

  it('StackPicker init value', () => {
    const wrapper = mount(
      <Picker.Stack
        dataSource={District}
        value={['340000', '340800', '340803']}
        />
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('StackPicker changeValue', () => {
    const onChangeFn = jest.fn();
    const wrapper = mount(
      <Picker.Stack
        dataSource={District}
        displayRender={onChangeFn}
        />
    );

    wrapper.find('.za-picker-stack-column').at(0).simulate('click');
    wrapper.find('.za-picker-stack-item').at(0).simulate('click');
    expect(onChangeFn).toBeCalled();
  });

  it('StackPicker maskClick', () => {
    const wrapper = mount(
      <Picker.Stack
        dataSource={District}
        />
    );
    wrapper.find('.za-picker-input').simulate('click');
    wrapper.find('.za-picker-cancel').simulate('click');
    wrapper.find('.za-mask').simulate('click');
  });
});
