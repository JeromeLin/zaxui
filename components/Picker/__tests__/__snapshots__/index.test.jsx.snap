// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Picker Cascader Picker 1`] = `
<Picker
  cancelText="取消"
  dataSource={
    Array [
      Object {
        "children": Array [
          Object {
            "label": "选项一",
            "value": "11",
          },
          Object {
            "label": "选项二",
            "value": "12",
          },
        ],
        "label": "选项一",
        "value": "1",
      },
      Object {
        "children": Array [
          Object {
            "label": "选项一",
            "value": "21",
          },
          Object {
            "label": "选项二",
            "value": "22",
          },
        ],
        "label": "选项一",
        "value": "2",
      },
    ]
  }
  disabled={false}
  displayAddon=""
  displayMember="label"
  okText="确定"
  onCancel={[Function]}
  onChange={[Function]}
  onClick={[Function]}
  onMaskClick={[Function]}
  onOk={[Function]}
  placeholder="请选择"
  prefixCls="za-picker"
  title="请选择"
  valueMember="value"
  visible={false}
>
  <div
    className="za-picker"
    onClick={[Function]}
  >
    <div
      className="za-picker-input za-picker-placeholder"
    >
      <input
        type="hidden"
        value={Array []}
      />
      请选择
    </div>
    <div
      className="za-picker-container"
      onClick={[Function]}
    >
      <Popup
        animationDuration={200}
        autoClose={false}
        className="za-popup-inner"
        direction="bottom"
        duration={3000}
        mask={true}
        maskType="normal"
        onClose={[Function]}
        onMaskClick={[Function]}
        prefixCls="za-popup"
        visible={false}
      >
        <div
          className="za-popup za-popup-hidden za-popup-inner"
        >
          <div
            className="za-popup-wrapper za-popup-wrapper-bottom"
            style={
              Object {
                "WebkitTransitionDuration": "200ms",
                "transitionDuration": "200ms",
              }
            }
          >
            <div
              className="za-picker-wrapper"
            >
              <div
                className="za-picker-header"
              >
                <div
                  className="za-picker-cancel"
                  onClick={[Function]}
                >
                  取消
                </div>
                <div
                  className="za-picker-title"
                >
                  请选择
                </div>
                <div
                  className="za-picker-submit"
                  onClick={[Function]}
                >
                  确定
                </div>
              </div>
              <div
                className="za-picker-mask-top"
              >
                <div
                  className="za-picker-mask-bottom"
                >
                  <Cascader
                    data={
                      Array [
                        Object {
                          "children": Array [
                            Object {
                              "label": "选项一",
                              "value": "11",
                            },
                            Object {
                              "label": "选项二",
                              "value": "12",
                            },
                          ],
                          "label": "选项一",
                          "value": "1",
                        },
                        Object {
                          "children": Array [
                            Object {
                              "label": "选项一",
                              "value": "21",
                            },
                            Object {
                              "label": "选项二",
                              "value": "22",
                            },
                          ],
                          "label": "选项一",
                          "value": "2",
                        },
                      ]
                    }
                    disabled={false}
                    displayMember="label"
                    onChange={[Function]}
                    prefixCls="za-picker"
                    value={Array []}
                    valueMember="value"
                  >
                    <ColumnGroup
                      displayMember="label"
                      onValueChange={[Function]}
                      prefixCls="za-picker"
                      selectedValue={
                        Array [
                          "1",
                          "11",
                        ]
                      }
                      valueMember="value"
                    >
                      <div
                        className="za-picker-column-group"
                      >
                        <div
                          className="za-picker-column-group-item"
                        >
                          <Column
                            displayMember="label"
                            onValueChange={[Function]}
                            prefixCls="za-picker"
                            pure={true}
                            selectedValue="1"
                            valueMember="value"
                          >
                            <div
                              className="za-picker-column"
                            >
                              <div
                                className="za-picker-column-indicator"
                              />
                              <div
                                className="za-picker-column-content"
                              >
                                <div
                                  className="za-picker-column-item za-picker-column-item-selected"
                                >
                                  选项一
                                </div>
                                <div
                                  className="za-picker-column-item"
                                >
                                  选项一
                                </div>
                              </div>
                            </div>
                          </Column>
                        </div>
                        <div
                          className="za-picker-column-group-item"
                        >
                          <Column
                            displayMember="label"
                            onValueChange={[Function]}
                            prefixCls="za-picker"
                            pure={true}
                            selectedValue="11"
                            valueMember="value"
                          >
                            <div
                              className="za-picker-column"
                            >
                              <div
                                className="za-picker-column-indicator"
                              />
                              <div
                                className="za-picker-column-content"
                              >
                                <div
                                  className="za-picker-column-item za-picker-column-item-selected"
                                >
                                  选项一
                                </div>
                                <div
                                  className="za-picker-column-item"
                                >
                                  选项二
                                </div>
                              </div>
                            </div>
                          </Column>
                        </div>
                      </div>
                    </ColumnGroup>
                  </Cascader>
                </div>
              </div>
            </div>
          </div>
          <Mask
            className=""
            onClose={[Function]}
            prefixCls="za-mask"
            style={
              Object {
                "WebkitAnimationDuration": "200ms",
                "animationDuration": "200ms",
              }
            }
            type="normal"
            visible={false}
          />
        </div>
      </Popup>
    </div>
  </div>
</Picker>
`;

exports[`Picker Picker 1`] = `
<div
  class="za-picker"
>
  <div
    class="za-picker-input za-picker-placeholder"
  >
    <input
      type="hidden"
      value=""
    />
    请选择
  </div>
  <div
    class="za-picker-container"
  >
    <div
      class="za-popup za-popup-hidden za-popup-inner"
    >
      <div
        class="za-popup-wrapper za-popup-wrapper-bottom"
        style="-webkit-transition-duration:200ms;transition-duration:200ms;"
      >
        <div
          class="za-picker-wrapper"
        >
          <div
            class="za-picker-header"
          >
            <div
              class="za-picker-cancel"
            >
              取消
            </div>
            <div
              class="za-picker-title"
            >
              请选择
            </div>
            <div
              class="za-picker-submit"
            >
              确定
            </div>
          </div>
          <div
            class="za-picker-mask-top"
          >
            <div
              class="za-picker-mask-bottom"
            >
              <div
                class="za-picker-column-group"
              >
                <div
                  class="za-picker-column-group-item"
                >
                  <div
                    class="za-picker-column"
                  >
                    <div
                      class="za-picker-column-indicator"
                    />
                    <div
                      class="za-picker-column-content"
                    >
                      <div
                        class="za-picker-column-item za-picker-column-item-selected"
                      >
                        选项一
                      </div>
                      <div
                        class="za-picker-column-item"
                      >
                        选项二
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Picker StackPicker 1`] = `
<PickerStack
  cols={Infinity}
  dataSource={
    Array [
      Object {
        "children": Array [
          Object {
            "label": "海淀区",
            "value": "11",
          },
          Object {
            "label": "西城区",
            "value": "12",
          },
        ],
        "label": "北京市",
        "value": "1",
      },
      Object {
        "children": Array [
          Object {
            "label": "黄埔区",
            "value": "21",
          },
          Object {
            "label": "虹口区",
            "value": "22",
          },
        ],
        "label": "上海市",
        "value": "2",
      },
    ]
  }
  disabled={false}
  displayItems={8}
  displayRender={[Function]}
  itemHeight={50}
  itemRender={[Function]}
  labelAddon=" > "
  onCancel={[Function]}
  onOk={[Function]}
  placeholder="请选择"
  prefixCls="za-picker"
  title="请选择"
  validate={[Function]}
  value={Array []}
  valueMember="value"
>
  <div
    className="za-picker"
  >
    <div
      className="za-picker-input za-picker-placeholder"
      onClick={[Function]}
    >
      <input
        type="hidden"
        value={Array []}
      />
      请选择
    </div>
    <div
      className="za-picker-container za-picker-hidden"
      onClick={[Function]}
    >
      <Popup
        animationDuration={200}
        autoClose={false}
        className={null}
        direction="bottom"
        duration={3000}
        mask={true}
        maskType="normal"
        onClose={[Function]}
        onMaskClick={[Function]}
        prefixCls="za-popup"
        visible={false}
      >
        <div
          className="za-popup za-popup-hidden"
        >
          <div
            className="za-popup-wrapper za-popup-wrapper-bottom"
            style={
              Object {
                "WebkitTransitionDuration": "200ms",
                "transitionDuration": "200ms",
              }
            }
          >
            <div
              className="za-picker-wrapper"
            >
              <div
                className="za-picker-header"
              >
                <div
                  className="za-picker-cancel"
                  onClick={[Function]}
                >
                  取消
                </div>
                <div
                  className="za-picker-title"
                >
                  请选择
                </div>
                <div
                  className="za-picker-submit"
                />
              </div>
              <div
                className="za-picker-crumbs"
              >
                <p>
                  选择：
                </p>
              </div>
              <div
                className="za-picker-stack-group"
              >
                <div
                  className="za-picker-stack-column lower-hidden"
                  onClick={[Function]}
                >
                  <div
                    className="za-picker-stack-column-wrapper"
                    onClick={[Function]}
                  >
                    <div
                      className="za-picker-stack-item"
                      onClick={[Function]}
                    >
                      北京市
                    </div>
                    <div
                      className="za-picker-stack-item"
                      onClick={[Function]}
                    >
                      上海市
                    </div>
                  </div>
                </div>
                <div
                  className="za-picker-stack-column lower-hidden"
                  onClick={[Function]}
                >
                  <div
                    className="za-picker-stack-column-wrapper"
                    onClick={[Function]}
                  >
                    <div
                      className="za-picker-stack-item"
                      onClick={[Function]}
                    >
                      海淀区
                    </div>
                    <div
                      className="za-picker-stack-item"
                      onClick={[Function]}
                    >
                      西城区
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <Mask
            className=""
            onClose={[Function]}
            prefixCls="za-mask"
            style={
              Object {
                "WebkitAnimationDuration": "200ms",
                "animationDuration": "200ms",
              }
            }
            type="normal"
            visible={false}
          />
        </div>
      </Popup>
    </div>
  </div>
</PickerStack>
`;
